### DISCUSSION TITLE
What's VarBuilder?

### DISCUSSION THREAD

**Discussion by shi-yan** [huggingface/candle#2076] [Q&A] [answered]
What's VarBuilder and why do I need it?

```rust
    let varmap = VarMap::new();
    let vs = VarBuilder::from_varmap(&varmap, DType::F32, dev);
```
saw it from

https://github.com/evgenyigumnov/candle-simplified-example

**Comment by EricLBuehler** [comment]
At a high level, a VarBuilder manages the parameters of your model. In that sense, it is equivalent to how Pytorch tracks your parameters automatically. At a low level the VarBuilder holds all the Tensors which are accessed by using `.get_with_hints`. You can "cd" by changing the prefix of the VarBuilder using `.pp`.

**Reply by shi-yan** [reply]
so it's the manger for model weights?

**Reply by EricLBuehler** [reply, marked as answer]
Yes, it is.
